VocabularyTrainer/Home Screen/HomeLanguageHeaderView.swift

```diff
@@ -0,0 +1,98 @@
+//
+
+import UIKit
+
+protocol HomeLanguageHeaderViewDelegate: AnyObject {
+    func tappedPracticeButton()
+    func tappedEditButton()
+    func tappedAddLanguageButton()
+}
+
+final class HomeLanguageHeaderView: UIView {
+
+    typealias Strings = HomeViewModel.Strings
+    typealias Colors = HomeViewModel.Colors
+
+    /// Title with trailing add button.
+    private let titleLabel = UILabel()
+    /// Button for adding a new language.
+    lazy var addLanguageButton: UIButton = {
+        let button = UIButton(type: .contactAdd)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.tintColor = .label
+        button.addAction(.init(handler: { [weak self] _ in self?.delegate?.tappedAddLanguageButton() }), for: .touchUpInside)
+        return button
+    }()
+    /// Button for starting practicing mode.
+    lazy var practiceButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        let text = NSAttributedString(string: Strings.practiceButtonTitle,
+                                      attributes: [.underlineStyle: NSUnderlineStyle.single.rawValue] )
+        button.setAttributedTitle(text, for: .normal)
+        button.tintColor = Colors.flippyGreen
+        button.titleLabel?.font = .preferredFont(forTextStyle: .body)
+        button.addAction(.init(handler: { [weak self] _ in self?.delegate?.tappedPracticeButton() }), for: .touchUpInside)
+        return button
+    }()
+    /// Button for editing a language.
+    lazy var editButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setTitle(Strings.editButtonTitle, for: .normal)
+        button.tintColor = .label
+        button.addAction(.init(handler: { [weak self] _ in self?.delegate?.tappedEditButton() }), for: .touchUpInside)
+        return button
+    }()
+    /// Stack view horizontally aligning `practiceButton` and `editButton`.
+    lazy var buttonStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [practiceButton, editButton])
+        stackView.spacing = Layout.defaultMargin
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+
+    weak var delegate: HomeLanguageHeaderViewDelegate?
+
+    // MARK: - Initializer
+
+    init() {
+        super.init(frame: .zero)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) { nil }
+
+    // MARK: - Setup
+
+    private func setupView() {
+        titleLabel.font = UIFontMetrics(forTextStyle: .title2).scaledFont(for: .systemFont(ofSize: 20, weight: .semibold))
+        titleLabel.text = Strings.headerTitle
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        translatesAutoresizingMaskIntoConstraints = false
+
+        addSubviews([titleLabel, addLanguageButton, buttonStackView])
+
+        NSLayoutConstraint.activate([
```

Comment:
Suggestion: isolate the constraints setup in a method.

Reply:
makes sense! updated here: 3562f2d8fdb5f668d44473b564f64b5f216d0bac